codec-http2/src/main/java/io/netty/handler/codec/http2/Http2HeadersMessage.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+public class Http2HeadersMessage extends Http2StreamMessage {
+    public final Http2Headers headers;
+    public final boolean endStream;
+    public final int padding;
+
+    /**
+     * Equivalent to {@code new Http2HeadersMessage(headers, false)}.
+     *
+     * @param headers the non-{@code null} headers to send
+     */
+    public Http2HeadersMessage(Http2Headers headers) {
+        this(headers, false);
+    }
+
+    /**
+     * Equivalent to {@code new Http2HeadersMessage(headers, endStream, 0)}.
+     *
+     * @param headers the non-{@code null} headers to send
+     */
+    public Http2HeadersMessage(Http2Headers headers, boolean endStream) {
+        this(headers, endStream, 0);
+    }
+
+    /**
+     * Construct a new headers message.
+     *
+     * @param headers the non-{@code null} headers to send
+     * @param endStream whether these headers should terminate the stream
+     * @param padding additional bytes that should be added to obscure the true content size
+     */
+    public Http2HeadersMessage(Http2Headers headers, boolean endStream, int padding) {
+        this.headers = checkNotNull(headers, "headers");
+        this.endStream = endStream;
+        if (padding < 0) {
```

Comment:
Should we have a `static` method to check this? For example do we need to also check upper bound?  Either way I think this is a common check across a few places...the existing codec probably should have done this already.